Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -1083,6 +1083,18 @@ public class PBXProjGenerator {
             }
         }
 
+        func addresourcesBuildPhase() {
+            let resourcesBuildPhaseFiles = getBuildFilesForPhase(.resources) + copyResourcesReferences
+            if !resourcesBuildPhaseFiles.isEmpty {
+                let resourcesBuildPhase = addObject(PBXResourcesBuildPhase(files: resourcesBuildPhaseFiles))
+                buildPhases.append(resourcesBuildPhase)
+            }
+        }
+
+        if target.resourcesBeforeSourcesBuildPhase {
```

Comment:
This will add the `PBXResourcesBuildPhase` before the `PBXSourcesBuildPhase` if the `resourcesBeforeSourcesBuildPhase` property is true. I've moved the logic for adding the `PBXResourcesBuildPhase` to a function to not duplicate it.